src/components/common/Accordion/BasicAccordion.tsx

```diff
@@ -0,0 +1,242 @@
+    tab: number;
+    add?: string;
+}
+
+export const buttonStyles = {
+    fontSize: '1rem',
+    borderRadius: '0',
+    textTransform: 'none',
+    marginRight: '20px',
+};
+export const BasicAccordion = ({ url, tab, add }: Props) => {
+    const { searchValue, setSearchValue } = useContext(SearchValueContext);
+    const { params, setParams } = useContext(FilterDataContext);
+    const [currentPage, setCurrentPage] = useState(1);
+    const [rowsPerPage, setRowsPerPage] = useState(10);
+    const [totalCount, setTotalCount] = useState(0);
+    const [totalPages, setTotalPages] = useState(0);
+    const [users, setUsers] = useState<UserAndSkills[]>([]);
+    const [loading, setLoading] = useState(true);
+    const [expanded, setExpanded] = useState<string | false>(false);
+    const { id } = useParams();
+
+    const {
+        snackBarMessage,
+        snackBarType,
+        isSnackBarOpen,
+        hideSnackBar,
+        showSnackBar,
+    } = useSnackBar();
+    useEffect(() => {
+        const fetchAvailableUsers = async (url: string) => {
+            try {
+                const urlProps = `${url}?page=${currentPage}&limit=${rowsPerPage}`;
+                const urlSearch = `${apiUrl}${ENDPOINTS.search}/${id}/?search=${searchValue}&tab=${tab}`;
+                const urlFilter = `${apiUrl}${ENDPOINTS.filter}/${id}/?${params}&tab=${tab}`;
+                const availableUrl = searchValue ? urlSearch : (params ? urlFilter : urlProps);
+                const response = await axios.get(availableUrl, { withCredentials: true });
+                const { users, totalCount, totalPages } = response.data;
+                setUsers(users);
+                setTotalCount(totalCount);
+                if (totalCount === 0) {
+                    showSnackBar('Brak kursantów', SnackBarEnum.ERROR_MESSAGE);
+                }
+                setTotalPages(totalPages);
+                setLoading(false);
+            } catch (error: any) {
+                setLoading(false)
+                handleErrorResponse(error, showSnackBar);
+            }
+        };
+        setLoading(true);
+        fetchAvailableUsers(url);
+    }, [currentPage, rowsPerPage, searchValue, params]);
+
+    const navigate = useNavigate();
+    const handleShowCV = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>, email?: string) => {
+        navigate(`/cv/${id}/${email}`);
+    };
+    const handleAction = async (event: React.MouseEvent<HTMLButtonElement, MouseEvent>, email?: string, status?: string) => {
+        event.stopPropagation();
+        try {
+            await axios.patch(
+                `${apiUrl}${ENDPOINTS.setStatus}/${id}`,
+                {
+                    email: email,
+                    status: status,
+                },
+                { withCredentials: true },
+            );
+            setUsers((prevUsers) => prevUsers.filter((user) => user.email !== email));
+            switch (status) {
+                case 'W trakcie rozmowy':
```

Comment:
Rozumiem, że to przychodzi  zenuma? Jeżeli tak to znacznie lepiej porównywać enumy.